2_preference_alignment/README.md

```diff
@@ -12,35 +12,29 @@ Alternative approaches like ORPO combine instruction tuning and preference align
+ORPO introduces a combined approach to instruction tuning and preference alignment in a single process. It modifies the standard language modeling objective by combining negative log-likelihood loss with an odds ratio term on a token level. The approach features a unified single-stage training process, reference model-free architecture, and improved computational efficiency. ORPO has shown impressive results across various benchmarks, demonstrating better performance on AlpacaEval compared to traditional methods. For more details, you can refer to the [Odds Ratio Preference Optimization (ORPO) documentation](./orpo.md).
 
-Key innovations:
-- Unified single-stage training process
-- Reference model-free architecture
-- Improved computational efficiency
+## Exercise Notebooks
 
-ORPO has shown impressive results across various benchmarks. Better performance on AlpacaEval compared to traditional methods. Strong results on MT-Bench, even without multi-turn training. Effective across different model sizes (125M to 1.3B parameters).
+| Title | Description | Exercise | Link | Colab |
+|-------|-------------|----------|------|-------|
+| DPO Training | Learn how to train models using Direct Preference Optimization | 🐢 Train a model using the Anthropic HH-RLHF dataset<br>🐕 Use your own preference dataset<br>🦁 Experiment with different preference datasets and model sizes | [Notebook](./notebooks/dpo_training_example.ipynb) | <a target="_blank" href="https://colab.research.google.com/github/huggingface/smol-course/blob/main/2_preference_alignment/notebooks/dpo_training_example.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a> |
+| ORPO Training | Learn how to train models using Odds Ratio Preference Optimization | 🐢 Train a model using instruction and preference data<br>🐕 Experiment with different loss weightings<br>🦁 Compare ORPO results with DPO | [Notebook](./notebooks/orpo_training_example.ipynb) | <a target="_blank" href="https://colab.research.google.com/github/huggingface/smol-course/blob/main/2_preference_alignment/notebooks/orpo_training_example.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a> |
 
-## Next Steps
-
-1. Start with [DPO](./dpo.md) for a simpler introduction to preference alignment.
-2. Explore [ORPO](./orpo.md) for a unified approach
-3. Try the practical tutorials in the `notebooks/` directory
 
 ## Resources
+
 - [TRL Documentation](https://huggingface.co/docs/trl/index)
 - [DPO Paper](https://arxiv.org/abs/2305.18290)
 - [ORPO Paper](https://arxiv.org/abs/2402.01714)
 - [Argilla RLHF Guide](https://argilla.io/blog/mantisnlp-rlhf-part-8/)
-
+- [Blog post on DPO](https://huggingface.co/blog/dpo-trl)
+- [TRL example script on DPO](https://github.com/huggingface/trl/blob/main/examples/scripts/dpo.py)
+- [TRL example script on ORPO](https://github.com/huggingface/trl/blob/main/examples/scripts/orpo.py)
```

Comment:
Maybe link the handbook too? As it contains recipes based on TRL for popular fine-tunes such as Zephyr

```
- [Hugging Face Alignment Handbook](https://github.com/huggingface/alignment-handbook)
```